pytorch/pytorch_checkpoint.py

```diff
@@ -84,9 +84,10 @@ def objective(trial):
     lr = trial.suggest_float("lr", 1e-5, 1e-1, log=True)
     optimizer = getattr(optim, optimizer_name)(model.parameters(), lr=lr)
 
-    if "checkpoint_path" in trial.user_attrs:
-        checkpoint_path = trial.user_attrs["checkpoint_path"]
-        trial_number = trial.user_attrs["trial_number"]
+    if "failed_trial" in trial.system_attrs:
+        trial_number = trial.system_attrs["failed_trial"]
+        trial_checkpoint_dir = os.path.join(CHECKPOINT_DIR, str(trial_number))
+        checkpoint_path = os.path.join(trial_checkpoint_dir, "model.pt")
         checkpoint = torch.load(checkpoint_path)
```

Comment:
Could you add checking the existence of the checkpoint? If the previous trial failed before the first model saving, this line raises `FileNotFoundError`.

Reply:
Added to scope, thanks.